ui/src/app/base/components/SubnetSelect/SubnetSelect.test.tsx

```diff
@@ -123,4 +123,25 @@ describe("SubnetSelect", () => {
       },
     ]);
   });
+
+  it("can filter the subnets using a provided function", () => {
+    const store = mockStore(state);
+    const wrapper = mount(
+      <Provider store={store}>
+        <Formik initialValues={{ subnet: "" }} onSubmit={jest.fn()}>
+          <SubnetSelect
+            filterFunction={(subnet) => subnet.vlan === 4}
```

Comment:
Might make this test clearer and easier to understand at a glance if we changed vlan to cidr as this is what we're testing below.
```suggestion
            filterFunction={(subnet) => subnet.cidr === "172.16.2.0/24"}
```